src/main/java/com/demo/classroom/Entity/Book.java

```diff
@@ -0,0 +1,29 @@
+package com.demo.classroom.Entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Data
+@NoArgsConstructor
```

Comment:
We are using these annotations so that we don't have to write Getter and Setter methods in our Model class. Spring provides lombok dependency which takes care of this for us.

Reply:
I meant `@NoArgsConstructor`. I was wondering if we have any requirements to create an entity without any arguments.

Reply:
If we don't have include any construct in our java class. Then Java compiler will automatically create a No Argument Constructor for us, which is known as **Default** constructor. 

But if we create any constructor(**with or without argument**) in our class then the Default Constructor won't be created. If we need to create an object without argument for example `Hard-Coded Admin Class` we must explicitly define a **no-argument constructor.** To resolve this we can use `@NoArgsConstructor` in our class.